apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -248,18 +249,40 @@ export const onSaveNoop =
   };
 
 // Thunk called when a save has failed.
-export const onSaveFail = () => (dispatch: AppDispatch) => {
-  dispatch(
-    addNotification({
-      id: getNewMessageId(),
-      text: 'Error updating project. Please try again.',
-      notificationType: 'error',
-      timestamp: Date.now(),
-    })
-  );
-  // Notify the UI that the save is complete.
-  dispatch(endSave());
-};
+export const onSaveFail =
+  (e: NetworkError) => (dispatch: AppDispatch, getState: () => RootState) => {
+    e.response.json().then(body => {
```

Comment:
One thing to check - does `response.json()` still work correctly (i.e. doesn't throw an error) for a different type of error, like a 500 internal server error?

Reply:
Oh - good call out! I added a catch to the promise chain to handle this and tested locally. Thanks!
![Screenshot 2024-06-28 at 4 20 31 PM](https://github.com/code-dot-org/code-dot-org/assets/107423305/ba56e76c-59d8-4aec-93ab-e66368812f25)